### DISCUSSION TITLE
Service Ceiling Limitations

### DISCUSSION THREAD

**Discussion by ICanDoThings2** [JSBSim-Team/jsbsim#624] [Q&A]
Does JSBSim have any kind of service ceilings for aircraft simulated? That is to say as with many aircraft after a certain altitude it becomes very difficult/slow to climb? If so, is it a hard value for aircraft, or one the simulation surmises based on input values, such as engine type, power, etc?

**Comment by seanmcleod** [comment]
There is no hard-coded value in JSBSim and there is no value specified in the aircraft's xml file, it simply comes down to the physics.

**Reply by ICanDoThings2** [reply]
I thought so. It does try to determine that though?

**Comment by seanmcleod** [comment]
No, why would JSBSim want to determine the service ceiling? It simply applies the basics physics on each time step.

**Reply by ICanDoThings2** [reply]
Alright. Well, for my purposes, service ceiling is of interest in a few ways, but this answers my question. Thank you.

**Comment by seanmcleod** [comment]
> Service ceiling is where the rate of climb drops below a prescribed value.
>
> The service ceiling is the maximum usable altitude of an aircraft. Specifically, it is the [density altitude](https://en.wikipedia.org/wiki/Density_altitude) at which flying in a [clean configuration](https://en.wikipedia.org/wiki/Clean_configuration), at the best [rate of climb](https://en.wikipedia.org/wiki/Rate_of_climb) [airspeed](https://en.wikipedia.org/wiki/Airspeed) for that altitude and with all engines operating and producing maximum continuous power, will produce a given rate of climb. A typical value might be 100 ft/min (0.51 m/s) climb,[[1]](https://en.wikipedia.org/wiki/Ceiling_%28aeronautics%29#cite_note-PHAK-1) or on the order of 500 ft/min (2.5 m/s) climb for jet aircraft.

Now if you actually wanted to calculate it for a particular JSBSim aircraft model you could use JSBSim for the "test flights", i.e. you could automate the process e.g. via a python script in order to generate the required data to generate a service ceiling graph like the one below.

![image](https://user-images.githubusercontent.com/824856/168649701-29956bc3-5d75-4090-8538-5e012a70cb04.png)

**Reply by ICanDoThings2** [reply]
I see! Thank you. But rather, I already know the service ceilings of the aircrafts I am trying to implement. I just wondered if the simulation would pick up on them or not. If it does, splendid! No action required on my end.

**Comment by seanmcleod** [comment]
If you have service ceiling data like the example above for the ATR 72 and you were trying to implement a JSBSim model of the ATR 72 then whether you'll see the exact same service ceiling when flying the model in JSBSim is really going to depend on how accurate your aerodynamic and engine data is.

JSBSim isn't going to by itself magically adapt the aerodynamic and engine data in order to match some test data you have like the graph above.

**Reply by ICanDoThings2** [reply]
Fair enough. That's what I assumed it would do though, have the same service ceiling with the correct data. In any case I'm sure it probably won't be that far off.

**Comment by dershow** [comment]
You should also be aware that the data you have could be some combination of aerodynamic and regulatory.  An aircraft might only be certified to a certain altitude due to oxygen requirements, pressurization system design, structural issues etc.  In that case, the aerodynamic would allow an aircraft to go higher than the regulations will.  In the example for the ATR-72 that Sean provided, the ceiling is a function of aircraft weight and temperature, but the limit is always 25,000 ft.  It is likely that the ATR-72 is not certified above that height, but from just the physics it can probably fly higher if it’s light and the air is cold.  JSBSim operates based on the physics, not the regulations.  So, make sure you understand what data you have for the aircraft.

--Adam

**Reply by ICanDoThings2** [reply]
Yes, that's a fine point. The aircraft I've been looking at lack pressurized cabins too, they're mostly WWII Aircraft so that may be why most have around the area of 10KM as service ceilings.
